File: sdk/python/src/agentpay_sdk/discovery.py
"""Discovery stub (mirrors sdk/typescript/src/discovery.ts).

MVP: always returns an empty list. Full semantic search arrives with
Discovery_Service (Task 13, post-MVP).
"""

from .client import AgentPayClient
from .types import AgentDiscoveryResult


def discover_agents(
    client: AgentPayClient,
    query: str | None = None,
    min_trust_score: int | None = None,
    limit: int | None = None,
) -> list[AgentDiscoveryResult]:
    """Discover agents (MVP stub — always returns [])."""
    return []

File: sdk/typescript/src/discovery.ts
import type { AgentPayClient } from "./client";
import type { AgentDiscoveryResult } from "./types";

/**
 * Discover agents (MVP stub — R11.1).
 *
 * In MVP, discovery is not yet implemented. This function always returns an
 * empty array. The Discovery_Service (Task 13, post-MVP) will provide the
 * full semantic search capability.
 *
 * @returns An empty array in MVP
 */
export async function discoverAgents(
  _client: AgentPayClient,
  _query?: string,
  _minTrustScore?: number,
  _limit?: number,
): Promise<AgentDiscoveryResult[]> {
  return [];
}
